Cookbook/Technical-Documents/UnitTestingViewModels.md

```diff
@@ -0,0 +1,51 @@
+## Unit testing view models
+
+### What do we test in the view model unit tests?
+
+- view model state changes
+- routing events
+- analytics events
+
+### Creating a new test case
+
+To create a test case for a view model use an Xcode template called `BentoViewModelTests` and provide the view model name (without `ViewModel` suffix). This template will create a basic structure for the test case that you will need to adjust to your specific implementation.
+
+### Writing a test
+
+When implementing a test use `perform(stub:when:assert:)` herlper method from `BabylonTestUtilities`.
```

Comment:
```suggestion
When implementing a test use `perform(stub:when:assert:)` helper method from `BabylonTestUtilities`.
```

Reply:
Maybe you could provide the full signature here? Because not having it memorised it's quite hard to follow (and this is aimed at new starters mostly, right?).

I think the reader should be able to have a more or less complete understanding of what is happening without having to look at the code in the project, and this doc only has truncated signatures and a non-structured description of parameters. I believe providing a complete, if short, example first, with full signatures so that the reader understands what's going on, and then dissecting it in following sections would be more conductive to learning.

Reply:
do you thin `perform<ViewModel: BoxViewModel>(stub: (TestScheduler) -> ViewModel, when: (ViewModel, TestScheduler) -> Void, assert: ([ViewModel.State]) -> Void)` will make it simpler to understand? I have doubts 😄